src/__mocks__/transaction-calldata.mock.ts

```diff
@@ -1,42 +1,31 @@
 import { faker } from '@faker-js/faker';
 import { ethers } from 'ethers';
+
 import {
-  getMultiSendCallOnlyDeployment,
-  getProxyFactoryDeployment,
-  getSafeSingletonDeployment,
-} from '@safe-global/safe-deployments';
-import type { DeploymentFilter } from '@safe-global/safe-deployments';
+  Gnosis_safe__factory,
+  Multi_send__factory,
+  Proxy_factory__factory,
+} from '../../contracts/safe/factories/v1.3.0';
+import { ERC20__factory } from '../../contracts/openzeppelin';
+import type { Gnosis_safeInterface } from '../../contracts/safe/v1.3.0/Gnosis_safe';
 
 export const MOCK_UNSUPPORTED_CALLDATA = new ethers.Interface([
   'function unsupported()',
 ]).encodeFunctionData('unsupported');
 
-function getSafeSingletonInterface(
-  filter?: DeploymentFilter,
-): ethers.Interface {
-  const safeDeployment = getSafeSingletonDeployment(filter);
-
-  if (!safeDeployment) {
-    throw new Error('Safe deployment not found');
-  }
-
-  return new ethers.Interface(safeDeployment.abi);
-}
-
 function getPrevalidatedSignature(address: string): string {
   return `${ethers.zeroPadValue(
     address,
     32,
   )}000000000000000000000000000000000000000000000000000000000000000001`;
 }
 
-function encodeSafeFunctionData(
-  ...args: Parameters<
-    InstanceType<typeof ethers.Interface>['encodeFunctionData']
-  >
+function withSafeInterface(
```

Comment:
This function seems quite complicated to me in comparison to just creating and directly using the interface.
i.e.
```ts
const safeInterface = Gnosis_safe__factory.createInterface();


export function getMockExecTransactionCalldata({
  to,
  value,
  data = '0x',
  operation = 0,
  safeTxGas = 0,
  baseGas = 0,
  gasPrice = 0,
  gasToken = faker.finance.ethereumAddress(),
  refundReceiver = faker.finance.ethereumAddress(),
  signatures = getPrevalidatedSignature(to),
}: {
  to: string;
  value: number;
  data?: string;
  operation?: 0 | 1;
  safeTxGas?: number;
  baseGas?: number;
  gasPrice?: number;
  gasToken?: string;
  refundReceiver?: string;
  signatures?: string;
}): string {
    return safeInterface.encodeFunctionData('execTransaction', [
      to,
      value,
      data,
      operation,
      safeTxGas,
      baseGas,
      gasPrice,
      gasToken,
      refundReceiver,
      signatures,
    ]);
}
```

Reply:
Good point. I've extracted the interfaces and fragments.